app/ml_models/rnn/train.py

```diff
@@ -0,0 +1,82 @@
+import torch.nn as nn
+import torch
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from app.ml_models.rnn.data_tools import WordLevelDataset
+from app.ml_models.rnn.rnn_model import RNNAnna
+from app.ml_models.rnn.generate import generate_word
+
+
+def train(rnn,
+          dataloader,
+          dataset,
+          learning_rate=0.0005,
+          epochs=500,
+          device='cpu',
+          name='straattaal',
+          save_every=50,
+          print_every=10000):
+
```

Comment:
```suggestion
def train(rnn,
          dataloader: torch.utils.data.DataLoader,
          dataset: torch.utils.data.Dataset,
          learning_rate: float = 0.0005,
          epochs: int = 500,
          device: str = 'cpu',
          name: str = 'straattaal',
          save_every: int = 50,
          print_every: int = 10000):
    """Train a full RNN model.
        
        Args:
            dataloader: Pytorch dataloader to iterate over the dataset.
            dataset: Dataset to be used for training and to generate a word.
            learning_rate: Learning rate to be used for training the model.
            epochs: Number of epochs to run our dataloader for.
            device: Device on which to run our model. Default is 'cpu', but can be set to GPU if one is available.
            name: Name to be used for the saved model.
            save_every: After every multitude of this number we save a version of the model.
            print_every: After every multitude of this number we print the loss.
        """
```

So we need the dataset for the generate_word function. I'm assuming it's going to be the same dataset as contained in the DataLoader object. Could we retrieve it from there instead to prevent some potential mismatch bugs?

Perhaps rename epochs to num_epochs.

Reply:
Yes I want to make a vocabulary object or something similar, this way you can easily map char to idx, idx to char, sample fro mthe vocabulary without having a full dataset object. Then we can set model.vocabulary = vocabulary ? or maybe save it in the statedict as well and load the vocabulary? what do you think @Sasafrass ?

 I will look at it soon

Reply:
I think a separate vocabulary object makes sense. Then we can move all the vocabulary specific functionality into one coherent class. It could probably live in either the ml_models/something or ml_models/rnn submodule. Could be done in a next PR.

Reply:
We would still need a way to load it as well, so we could either save it in the statedict or as a separate file? Any experience with which option is chosen more frequently in this case?

Reply:
In my limited experience, the vocabulary is stored separately. 
Multiple models can also share the same vocabulary this way. (i.e. all dutch models can share the mini vocabulary.txt file: it also becomes more efficient for larger vocabularies.).

#32